packages/nimble-components/src/select/index.ts

```diff
@@ -49,51 +146,689 @@ export class Select extends FoundationSelect implements ErrorPattern {
+    public override set value(next: string) {
+        const prev = `${this._value}`;
+        let newValue = next;
+
+        if (this.options?.length) {
+            const selectedIndex = this.options.findIndex(
+                el => el.value === newValue
+            );
+            const prevSelectedValue = this.options[this.selectedIndex]?.value ?? null;
+            const nextSelectedValue = this.options[selectedIndex]?.value ?? null;
+
+            if (
+                selectedIndex === -1
+                || prevSelectedValue !== nextSelectedValue
+            ) {
+                newValue = '';
+                this.selectedIndex = selectedIndex;
+            }
+
+            newValue = this.firstSelectedOption?.value ?? newValue;
+        }
+
+        if (prev !== newValue && !(this.open && this.selectedIndex < 0)) {
+            this._value = newValue;
+            super.valueChanged(prev, newValue);
+            if (!this.open) {
+                this.committedSelectedOption = this._options.find(
+                    o => o.value === newValue
+                );
+            }
+            Observable.notify(this, 'value');
+            if (this.collapsible) {
+                Observable.notify(this, 'displayValue');
+            }
+        }
+    }
+
+    public get displayValue(): string {
+        Observable.track(this, 'displayValue');
+        return this.committedSelectedOption?.text ?? '';
+    }
+
+    public regionChanged(
+        _prev: AnchoredRegion | undefined,
+        _next: AnchoredRegion | undefined
+    ): void {
+        if (this.region && this.control) {
+            this.region.anchorElement = this.control;
+        }
+    }
+
+    public controlChanged(
+        _prev: HTMLElement | undefined,
+        _next: HTMLElement | undefined
+    ): void {
+        if (this.region && this.control) {
+            this.region.anchorElement = this.control;
+        }
+    }
+
+    public setPositioning(): void {
         if (!this.$fastController.isConnected) {
             // Don't call setPositioning() until we're connected,
             // since this.forcedPosition isn't initialized yet.
             return;
         }
-        super.setPositioning();
+        const currentBox = this.getBoundingClientRect();
+        const viewportHeight = window.innerHeight;
+        const availableBottom = viewportHeight - currentBox.bottom;
+
+        if (this.forcedPosition) {
+            this.position = this.positionAttribute;
+        } else if (currentBox.top > availableBottom) {
+            this.position = SelectPosition.above;
+        } else {
+            this.position = SelectPosition.below;
+        }
+
+        this.positionAttribute = this.forcedPosition
+            ? this.positionAttribute
+            : this.position;
+
+        this.maxHeight = this.position === SelectPosition.above
+            ? Math.trunc(currentBox.top)
+            : Math.trunc(availableBottom);
```

Comment:
@jattasNI @rajsite, all of the above new implementation in this method. just copied directly from FAST.